File: tailSwap.js
function tailSwap(s) {
    let ss = s[0];
    let sss = s[1];
    let ssss = ss.split(':')[0];
    let sss$sss = ss.split(':')[1];
    let ssssSS = sss.split(':')[0];
    let SSS = sss.split(':')[1];
    return [`${ssss}:${SSS}`,`${ssssSS}:${sss$sss}`];
   }

console.log(tailSwap(['abc:123', 'cde:456']))
console.log(tailSwap(['a:12345', '777:xyz']))

File: tail_swap.py
import re
def tail_swap(strings):
    front = strings[0]
    back = strings[1]
    f1 = re.split(':', front)
    b1 = re.split(':', back)
    return [f1[0]+':'+b1[1], b1[0]+':'+f1[1]]

print(tail_swap(['abc:123', 'cde:456']))
print(tail_swap(['a:12345', '777:xyz']))